src/Models/Traits/ConfiguresSubscriptionGateway.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Payavel\Subscription\Models\Traits;
+
+use Payavel\Subscription\SubscriptionGateway;
+
+trait ConfiguresSubscriptionGateway
+{
+    /**
+     * The subscription's pre-configured gateway.
+     *
+     * @var \Payavel\Subscription\SubscriptionGateway
+     */
+    private $subscriptionGateway;
+
+    /**
+     * Retrieve the subscription's configured gateway.
+     *
+     * @return \Payavel\Subscription\SubscriptionGateway
+     */
+    public function getGatewayAttribute()
+    {
+        if (! isset($this->subscriptionGateway)) {
+            $this->subscriptionGateway = (new SubscriptionGateway);
```

Comment:
I will push the fix to be able to retrieve the provider with the config driver, and let you know when that is complete so you can update.

Reply:
Completed in https://github.com/payavel/subscription/pull/21. You may proceed on configuring the gateway based on `provider_id`.

Reply:
Done!